### ISSUE TITLE
pixelwise-loss weight map part 2

### ISSUE THREAD

**Issue by shinlin77** [tensorflow/tensorflow#38206]
<em>Please make sure that this is a feature request. As per our [GitHub Policy](https://github.com/tensorflow/tensorflow/blob/master/ISSUES.md), we only address code/doc bugs, performance issues, feature requests and build/installation issues on GitHub. tag:feature_template</em>

**System information**
- TensorFlow version (you are using): 2.0
- Are you willing to contribute it (Yes/No): Yes

**Describe the feature and the current behavior/state.**
This a continuation of 

https://github.com/tensorflow/tensorflow/issues/38060

which I think might have been prematurely closed.  It was said that tensorflow 2.x allows for pixelwise-loss map.  

However, I looked at  https://www.tensorflow.org/api_docs/python/tf/keras/Model#fit and pasted the relevant parts.  These options, I don't think, will be adequate for the weights needed for pixelwise-loss-weights.  For 2D images, weights will need to be shape [None, x_dim, y_dim, 1 or 3 or 4].  You see, the weights also need to be passed through  tf.keras.preprocessing.image. ImageDataGenerator along with X_train, Y_train.  So X_datagen, W_datagen (for weights), and Y_datagen all need all to be passed into Model.fit.  I don't see how to do that.

   - *class_weight*: Optional dictionary mapping class indices (integers)
   to a weight (float) value, used for weighting the loss function (during
   training only). This can be useful to tell the model to "pay more
   attention" to samples from an under-represented class.
   - *sample_weight*: Optional Numpy array of weights for the training
   samples, used for weighting the loss function (during training only). You
   can either pass a flat (1D) Numpy array with the same length as the input
   samples (1:1 mapping between weights and samples), or in the case of
   temporal data, you can pass a 2D array with shape (samples,
   sequence_length), to apply a different weight to every timestep of every
   sample. In this case you should make sure to specify
   sample_weight_mode="temporal" in compile(). This argument is not
   supported when x is a dataset, generator, or keras.utils.Sequence
   <https://www.tensorflow.org/api_docs/python/tf/keras/utils/Sequence> instance,
   instead provide the sample_weights as the third element of x.

**Will this change the current api? How?**
It will add more versatility to the api by incorporating an important aspect of computer vision.

**Who will benefit with this feature?**
Those using keras to implement computer vision programs.

**Any Other info.**
Thank you kindly for your consideration.

**Comment by shinlin77** [comment]
I am thrilled this request got to François Chollet.  I'm just wondering what the team's current thinking is with regard to incorporating pixelwise-loss weight maps.  It seems to me a veritable "must-have" for training deep learning algorithms for instance segmentation of objects that are often close together.

**Comment by Gibson-Clark** [comment]
Hello, @shinlin77 I have been following your question's about pixel-wise weight maps to be used in training and I too hope to follow Ronneberger, Fischer, and Brox's approach used in UNET.

I saw the workaround you linked [https://groups.google.com/g/keras-users/c/ue1S8uAPDKU/m/PzL7Vb0bBQAJ](url)

And I wanted to check in since then to to hear if anyone had another solution to this challenge in the interim?

**Comment by Gibson-Clark** [comment]
Thanks @shinlin77 for sharing. It is good to know that this is the approach that you landed on, I will be referring to the sample code.

Link corrected, thanks for that note.

**Comment by gowthamkpr** [comment]
@shinlin77 Is this still relevant? Can you please close this issue if it is not relevant. Thanks!
